src/modules/robot/arm_solutions/RotatableDeltaSolution.cpp

```diff
@@ -0,0 +1,305 @@
+	delta_printable_radius = config->value((unsigned int)delta_printable_radius_checksum)->by_default(150.0f)->as_number();
+
+	//While the by_default() method fills in default values if the field is missing from the config
+	//it does not prevent crazy numbers being inputed, so we try to prevent odd ratios and divide by zero
+	//errors in our configuration calculations
+	xyz_full_steps_per_rotation = config->value((unsigned int)xyz_full_steps_per_rotation_checksum)->by_default(200.0f)->as_number();
+	if (xyz_full_steps_per_rotation < 1) xyz_full_steps_per_rotation = 200.0; //try to prevent stupid config entries
+
+	xyz_microsteps = config->value((unsigned int)xyz_microsteps_checksum)->by_default(16.0f)->as_number();
+	if (xyz_microsteps < 1) xyz_microsteps = 16.0; //try to prevent stupid config entries
+
+	small_pulley_teeth = config->value((unsigned int)small_pulley_teeth_checksum)->by_default(16.0f)->as_number();
+	if (small_pulley_teeth < 1) small_pulley_teeth = 16.0; //try to prevent stupid config entries
+
+	big_pulley_teeth = config->value((unsigned int)big_pulley_teeth_checksum)->by_default(150.0f)->as_number();
+	if (big_pulley_teeth < 1) big_pulley_teeth = 150.0; //try to prevent stupid config entries
+
+	//
+	//We need to override the alpha_steps_per_mm, beta_steps_per_mm & gamma_steps_per_mm read from the config file
+	//as these must be calculated based on other parameters in the config file on the SD card.
+	//The formula is:
+	//     (xyz_full_steps_per_rotation*xyz_microsteps*(big_pulley_teeth/small_pulley_teeth))*(two_pi*delta_rf)
+	//
+	//which means:
+	//
+	//xyz_full_steps_per_rotation*xyz_microsteps gives the number of steps the stepper motor needs to rotate 360 degrees
+	//
+	//(big_pulley_teeth/small_pulley_teeth) gives the ratio of the large pulley to the small one so that we can determine
+	//the  number of rotations the stepper motor needs to rotate the large pulley fully 360 degrees
+	//
+	//and the two_pi*delta_rf gives the circumference of the large pulley itself.
+	//
+	//So the formula informs us the number of steps the stepper motor needs to move a mm on the large pulley.
+	//
+	//For example, a 0.9 degree stepper has 400 steps per revolution, driven using a 1/32 microstep DRV8825 driver chip
+	//with a 16 tooth small pulley on the stepper motor and a 150 tooth (equivalent) large pulley of radius 90mm results in:
+	//
+	//(400*32*(150/16))/(2 * pi * 90) = 212.206590789194 microsteps are required to move the effector attached to
+	//the large pulley 1mm in a linear sense.
+	//
+	//Lastly, if you are not using a small and large pulley system like the FirePick is, simply make these two values
+	//the same, BUT NOT ZERO.
+
+	init();
+}
+
+// inverse kinematics
+// helper functions, calculates angle theta1 (for YZ-pane)
+int RotatableDeltaSolution::delta_calcAngleYZ(float x0, float y0, float z0, float &theta)
+{
+    float y1 = -0.5 * tan30 * delta_f; // f/2 * tan 30
```

Comment:
All constants need to be appended with F otherwise all calculations are done in double as constants without F are double. Thi sneeds to be done in several lines following this one.